dbms/src/Dictionaries/ClickHouseDictionarySource.cpp

```diff
@@ -143,4 +159,19 @@ BlockInputStreamPtr ClickHouseDictionarySource::createStreamForSelectiveLoad(con
     return std::make_shared<RemoteBlockInputStream>(pool, query, sample_block, context);
 }
 
+std::string ClickHouseDictionarySource::doInvalidateQuery(const std::string & request) const
+{
+    Block invalidate_sample_block;
+
+    if (is_local)
+    {
+        auto input_block = executeQuery(request, context, true).in;
+        return readInvalidateQuery(dynamic_cast<IProfilingBlockInputStream&>((*input_block)));
+    }
+ 
+    auto invalidate_stream = RemoteBlockInputStream(pool, request, invalidate_sample_block, context);
```

Comment:
Why not `RemoteBlockInputStream invalidate_stream(pool, request, invalidate_sample_block, context);`?

Reply:
For my fun)